### PR TITLE
feat: Added user routes

### PR DESCRIPTION
**Pull request by AndrewKaenko**
Added user routes

### CODE REVIEW

File: src/services/user.service.ts [hunk 1]
```diff
@@ -1,9 +1,58 @@
...
-export const getAdmin = () => {
-  getAdminUser();
+export const getUser = async (id: string) => {
+  try {
+    const result = await getUserById(id);
+    if (result) {
+      return result;
+    }
+  } catch (error) {
+    throw new Error("Error");
```
Comment (NikPaukov) [inline src/services/user.service.ts]: should be more detalised errors)
```suggestion
    throw new Error(error.message);
```

File: src/services/user.service.ts [hunk 2]
```diff
@@ -1,9 +1,58 @@
...
+  getUserById,
+  updateOne,
+} from "../repositories/user.repository";
 
-export const getUserById = (id: string) => {
-  userRepository.findById(id);
+export const listUsers = async () => {
+  try {
+    const result = await getList();
+    if (result) {
```
Comment (NikPaukov) [inline src/services/user.service.ts]: whether i dont get it or it will return undefined if there is no result

**Review by NikPaukov — CHANGES_REQUESTED** [review]

File: src/repositories/user.repository.ts [hunk 3]
```diff
@@ -1,10 +1,63 @@
...
-  return UserModel.find({
-    username: "admin",
-  });
+export const getList = async () => {
+  try {
+    const result = await UserModel.find();
+    return result;
+  } catch (error) {
+    throw new Error("Error in the DB");
+  }
```
Comment (alexandrtovmach) [inline src/repositories/user.repository.ts]: don't use try catch

File: src/repositories/user.repository.ts [hunk 4]
```diff
@@ -1,10 +1,63 @@
...
+};
+
+export const updateOne = async (id: string, data: User) => {
+  try {
+    await UserModel.updateOne(
+      { _id: id },
+      { ...data, updated_at: new Date().toISOString() }
```
Comment (alexandrtovmach) [inline src/repositories/user.repository.ts]:
```suggestion
      { ...data, updatedAt: new Date().toISOString() }
```

**Review by alexandrtovmach — CHANGES_REQUESTED** [review]

File: src/models/User.model.ts [hunk 5]
```diff
@@ -3,6 +3,18 @@ import { model, Schema, Document } from "mongoose";
...
+  url: string;
+  description: string;
+  verified: boolean;
+  followers_count: number;
+  friends_count: number;
+  listed_count: number;
+  favourites_count: number;
+  statuses_count: number;
+  created_at: Date;
+  updated_at: Date | null;
```
Comment (VitaliiDenysiuk) [inline src/models/User.model.ts]:
```suggestion
  updated_at?: Date;
```

File: src/models/User.model.ts [hunk 6]
```diff
@@ -3,6 +3,18 @@ import { model, Schema, Document } from "mongoose";
 export interface User extends Document {
   username: string;
   password: string;
+  name: string;
+  location: string;
```
Comment (VitaliiDenysiuk) [inline src/models/User.model.ts]: Is this field required?
```suggestion
  location?: string;
```

File: src/models/User.model.ts [hunk 7]
```diff
@@ -33,6 +45,63 @@ const UserSchema: Schema = new Schema<User>(
...
+      required: false,
+      default: 0,
+    },
+    statuses_count: {
+      type: Number,
+      required: false,
+      default: 0,
+    },
+    created_at: {
+      type: Date,
```
Comment (VitaliiDenysiuk) [inline src/models/User.model.ts]:
```suggestion
      type: String,
```
you use toISOString() for this field

File: src/routes/users.route.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  try {
+    const result = await listUsers();
+    if (result) {
+      return res.send(result);
+    }
+  } catch (error) {
+    console.error(error);
+    return res.sendStatus(500);
+  }
+});
```
Comment (VitaliiDenysiuk) [inline src/routes/users.route.ts:16]: remove try ...catch too

**Review by VitaliiDenysiuk — CHANGES_REQUESTED** [review]

**Review by alexandrtovmach — APPROVED** [review]

**Pull request merged**